api/filters/patchstrategicmerge/patchstrategicmerge_test.go

```diff
@@ -921,3 +921,150 @@ data:
 		})
 	}
 }
+
+func TestFilterMergeOptionsListAppend(t *testing.T) {
```

Comment:
Thank you so much for your PR.
i would suggest considering the addition of some edge case tests:

- append to empty list
- append duplicate element

etc. IMHO